pkg/odo/cli/create/namespace/namespace.go

```diff
@@ -105,7 +108,28 @@ func (nco *NamespaceCreateOptions) Run(ctx context.Context) (err error) {
 	if err != nil {
 		return err
 	}
-
+	if nco.waitFlag {
+		var timeOut = time.After(nco.clientset.PreferenceClient.GetTimeout())
+	L:
+		for {
+			select {
+			case <-timeOut:
+				log.Warningf("Timeout while waiting for %s %q; it may take a while to appear", nco.commandName, nco.namespaceName)
+				return nil
+			default:
+				var nsList project.ProjectList
+				nsList, err = nco.clientset.ProjectClient.List()
+				if err != nil {
+					klog.V(4).Infof("Failed to list %ss", nco.commandName)
+				}
+				for _, ns := range nsList.Items {
+					if ns.Name == nco.namespaceName {
+						break L
+					}
+				}
```

Comment:
I would add a sleep for let's say 50ms at this point, to not respawn the List operation too fast

Reply:
That's a good idea.